packages/@aws-cdk/cdk/lib/physical-name.ts

```diff
@@ -0,0 +1,116 @@
+import { PhysicalNameGenerator } from "./physical-name-generator";
+import { IResource } from './resource';
+import { Token } from "./token";
+
+/**
+ * Options allowing customizing the automatically generated physical name of the resource.
+ * Used in the {@link PhysicalName#auto} method.
+ */
+export interface AutoPhysicalNameOptions {
+  /**
+   * Can the framework generate and set a physical name for the resource automatically
```

Comment:
Change docs to:

```
Allow the framework to automatically allocate a physical name during synthesis 
in case this resource is referenced across environments.

When resources are referenced across environments, the physical name of the 
resource must be known during synthesis in order to avoid the need to communicate
 across environments during deployment. This also removes the potential of cyclic 
references.

@default false A physical name will be allocated during deployment by AWS CloudFormation.
```

Reply:
Changed.